### PR TITLE
Build and test in same step in Windows CI

### PR DESCRIPTION
**Pull request by yuche**
Windows CI seems to purge files that have been created after each step, resulting in compiled files that cannot be found in the test.

This PR resolves the testing issue with #1800.

failed case:
https://dev.azure.com/nervosnetwork/neuron/_build/results?buildId=13509&view=logs&j=1dd1a9fa-5bd6-5a58-be75-4682264ffaf4&t=0fd07730-01dd-5646-2168-32456ff1e2aa&l=861

success case(in my own repo):
https://dev.azure.com/nervjs/Nervjs/_build/results?buildId=65&view=logs&j=1dd1a9fa-5bd6-5a58-be75-4682264ffaf4&t=0fd07730-01dd-5646-2168-32456ff1e2aa&l=232

### CODE REVIEW

File: azure-pipelines.yml [hunk 1]
```diff
@@ -83,10 +83,10 @@ stages:
         yarn bootstrap
         set CI=false
         yarn build
-      name: Build
+      name: Bootstrap
     - script: |
         set CI=true
-        yarn test
+        yarn test:ci
```
Comment (Keith-CY) [inline azure-pipelines.yml:89]: Thanks to this commit that I've found something interesting.

We're testing `*.ts` then why do we build before the tests?

File: azure-pipelines.yml [hunk 2]
```diff
@@ -83,10 +83,10 @@ stages:
         yarn bootstrap
         set CI=false
         yarn build
```
Comment (Keith-CY) [inline azure-pipelines.yml:85]: Thanks to this commit that I've found something interesting.

We're testing `*.ts` then why do we build before the tests?

Comment (yuche) [reply azure-pipelines.yml:85]: Now we have some test cases executing .js files via child_process.fork:
https://github.com/nervosnetwork/neuron/blob/7d1b00b58fda76e282cbc92206b1550caa36eb29/packages/neuron-wallet/tests/utils/worker.test.ts#L39

Comment (Keith-CY) [reply azure-pipelines.yml:85]: > Now we have some test cases executing .js files via child_process.fork:
> https://github.com/nervosnetwork/neuron/blob/7d1b00b58fda76e282cbc92206b1550caa36eb29/packages/neuron-wallet/tests/utils/worker.test.ts#L39

I was curious about adding the build step before. 🤔

Comment (katat) [reply azure-pipelines.yml:85]: @yuche 
Could you please explain a bit more about why we need to change the build script to accommodate this test suit?

**Comment by katat** [comment]
Looks like this test suit is for integration tests on `utils/worker`. If so, use the name `utils/worker.intg.test.ts`. We use the file extension `intg` to differentiate from unit tests.

File: azure-pipelines.yml (hunk 2 above)
Comment (katat) [reply azure-pipelines.yml:85]: Because it is a mock-up fixture, it is only for the test purpose. What if we directly save it as a `js` file for the fork function to refer to?

Comment (yuche) [reply azure-pipelines.yml:85]: > Because it is a mock-up fixture, it is only for the test purpose. What if we directly save it as a `js` file for the fork function to refer to?

We are using a pure javascript file for `child_process.fork`,  But that file needs to import the compiled file:

https://github.com/nervosnetwork/neuron/blob/7d1b00b58fda76e282cbc92206b1550caa36eb29/packages/neuron-wallet/tests/utils/fixtures/worker.js#L1-L6

To summarize, we need to test `child_process.fork`, but `child_process.fork` can only use .js files, and .js files can only import .js source code, so we have to compile the source code before testing.

Comment (katat) [reply azure-pipelines.yml:85]: Right. I mean can we just save that specific worker fixture file as a pure javascript file instead of a typescript, so no need to involve the compilation?

Comment (yuche) [reply azure-pipelines.yml:85]: > > Now we have some test cases executing .js files via child_process.fork:
> 
> I was curious about adding the build step before. 🤔




> Right. I mean can we just save that specific worker fixture file as a pure javascript file instead of a typescript, so no need to involve the compilation?

I'm sorry I didn't get that, but the worker fixture file is now a pure javascript file.

Comment (katat) [reply azure-pipelines.yml:85]: Ok, I assumed it was a ts file. 

How about using this technique. Does it work in our context? https://github.com/TypeStrong/ts-node/issues/619#issuecomment-511366883

Comment (yuche) [reply azure-pipelines.yml:85]: > Ok, I assumed it was a ts file.
> 
> How about using this technique. Does it work in our context? [TypeStrong/ts-node#619 (comment)](https://github.com/TypeStrong/ts-node/issues/619#issuecomment-511366883)

I haven't tested it yet, but this should work. But then we need to introduce an additional dependency: `ts-node`.

Perhaps I wasn't  describe it clear enough, but the current implementation actually runs on macOS and Linux, as well as Windows on the local machine, but the Windows instance of the Azure pipeline cleans up the files already created at each step, so the test fails.

Before this PR, the original procedure was to compile the source files before testing, but testing and compiling were in separate steps, except that this PR puts compiling and testing in one step in Windows CI.

Comment (katat) [reply azure-pipelines.yml:85]: Makes sense to me

**Review by katat — APPROVED** [review]

File: azure-pipelines.yml (hunk 2 above)
Comment (Keith-CY) [reply azure-pipelines.yml:85]: If I got it right, this commit is to fix a test failure introduced by a PR that is not merged yet? Then why not merge this fix into that PR?

Comment (Keith-CY) [reply azure-pipelines.yml:85]: And any documents indicate that `but the Windows instance of the Azure pipeline cleans up the files already created at each step, so the test fails.`? It is kinda weird since Mac and Linux don't do that.

Comment (yuche) [reply azure-pipelines.yml:85]: > And any documents indicate that `but the Windows instance of the Azure pipeline cleans up the files already created at each step, so the test fails.`? It is kinda weird since Mac and Linux don't do that.

No, but this PR does fix the problem, and I've tested it in my own repo. Please check out the two links in the PR description.

Comment (Keith-CY) [reply azure-pipelines.yml:85]: > > And any documents indicate that `but the Windows instance of the Azure pipeline cleans up the files already created at each step, so the test fails.`? It is kinda weird since Mac and Linux don't do that.
> 
> No, but this PR does fix the problem, and I've tested it in my own repo. Please check out the two links in the PR description.

Don't mind that, I was just curious about that.

Comment (Keith-CY) [reply azure-pipelines.yml:85]: But [this](https://github.com/nervosnetwork/neuron/pull/1811#discussion_r472677882) is my question, why not merge this into the PR introduced the test failure.

Comment (yuche) [reply azure-pipelines.yml:85]: > But [this](https://github.com/nervosnetwork/neuron/pull/1811#discussion_r472677882) is my question, why not merge this into the PR introduced the test failure.

I don't have access to the Neuron repository's branch protection rules, but I'm guessing we're supposed to be protecting the develop branch. since #1800 merge to develop branch, so we need to merge this PR into the develop branch for the CI to work.

Comment (Keith-CY) [reply azure-pipelines.yml:85]: I mean merge this commit into https://github.com/nervosnetwork/neuron/tree/refactor/upgrade-to-electron-9

**Review by Keith-CY — APPROVED** [review]

File: azure-pipelines.yml (hunk 2 above)
Comment (Keith-CY) [reply azure-pipelines.yml:85]: It seems that this branch including extra commits to https://github.com/nervosnetwork/neuron/tree/refactor/upgrade-to-electron-9, I set the pr base back to develop
